### ISSUE TITLE
Only open one model of a file at a time

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#4201]
Currently we can, for example, open a notebook as both a text file and a notebook, but those use different underlying models to represent the file. As such, edits in one can cause conflicts in the other, which are only discovered when saving.

Should we only allow one model of a file to be active at a time? If you try to open a widget that uses a different model, perhaps there should be a message saying that all open instances of the file must be closed before the new open, with a button helpfully offering to close all views of that particular file.

**Comment by ian-r-rose** [comment]
I agree that the current behavior can result in confusion. That being said, I am having a hard time envisioning a dialog that accurately represents why you can open multiple views for a document in some cases, but not in others. How do you think we can do this without making things *more* confusing to the user?
